addons/website_sale/models/product_template.py

```diff
@@ -135,6 +143,11 @@ def _default_website_sequence(self):
 
     #=== COMPUTE METHODS ===#
 
+    @api.depends('is_published')
+    def _compute_publish_date(self):
+        """Set `publish_date` to the moment of (re-)publishing."""
+        self.filtered('is_published').publish_date = fields.Datetime.now()
```

Comment:
To the best of my knowledge, I think we should always assign a value (even without change) to all records in a compute method.

Reply:
Didn't know that used to be a thing. The compute method right below doesn't do that either, so I'm sure it's fine https://github.com/odoo/odoo/blob/a363da657513c076b97320b1776594eb92311b37/addons/website_sale/models/product_template.py#L151-L156

Reply:
No, it would be more verbose and cluttered if the logic that can be computes would be moved there.